plugins/auth-backend-module-guest-provider/src/authenticator.ts

```diff
@@ -16,23 +16,60 @@
 
 import { createProxyAuthenticator } from '@backstage/plugin-auth-node';
 import { NotAllowedError } from '@backstage/errors';
+import { stringifyEntityRef, UserEntity } from '@backstage/catalog-model';
+
+const DEFAULT_USER_ENTITY_REF = stringifyEntityRef({
+  kind: 'user',
+  namespace: 'development',
+  name: 'guest',
+});
 
 export const guestAuthenticator = createProxyAuthenticator({
-  defaultProfileTransform: async () => {
-    return { profile: {} };
+  defaultProfileTransform: async (
+    result: { userEntityRef?: string },
+    context,
+  ) => {
+    // Get the userEntityRef from the result (passed from authenticate function)
+    const userEntityRef = result.userEntityRef || DEFAULT_USER_ENTITY_REF;
+
+    try {
+      // Try to find the user in the catalog
+      const catalogUser = await context.findCatalogUser({
```

Comment:
It is a bit unfortunate to be forced to call the catalog twice (once in the resolver and once in here) - is there nothing we can do about that? 🤔 It's no big deal really and sign-ins don't happen all that often, but wanted to check